awsbestpractices/custom-retries/CustomRetries.java

```diff
@@ -0,0 +1,120 @@
+package example;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.lambda.AWSLambda;
+import com.amazonaws.services.lambda.model.InvokeRequest;
+
+import java.io.IOException;
+
+public class Example implements RequestHandler<ScheduledEvent, Void> {
+
+    private AWSLambda lambdaClient;
+    private MyOtherClient otherClient;
+
+    // Should flag
+    public void retry() {
+        final InvokeRequest request = new InvokeRequest();
+        while (true) {
+            try {
+                return lambdaClient.invoke(request);
+            } catch (AmazonServiceException e) {
+            }
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 16:22, 38:44, 27:33.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.